byron/ledger/executable-spec/test/Generator.hs

```diff
@@ -0,0 +1,109 @@
+module Generator where
+
+import           Data.Bits (shiftL)
+import           Data.Word (Word64)
+import           Hedgehog
+import qualified Hedgehog.Gen    as Gen
+import qualified Hedgehog.Range  as Range
+import           Ledger.Core (Slot(..), SlotCount(..), BlockCount(..))
+import           Ledger.GlobalParams (k)
+import           Ledger.Update (ProtVer(..), PParams(..))
+import           Numeric.Natural (Natural)
+
+-- | Generates a slot within the given bound
+slotGen :: Word64 -> Word64 -> Gen Slot
+slotGen lower upper = do
+  Slot <$> Gen.word64 (Range.linear lower upper)
+
+-- | Generates a protocol version
+protVerGen :: Gen ProtVer
+protVerGen = do
+  pvMin  <- Gen.integral (Range.linear (0 :: Natural) 100)
+  pvDiff <- Gen.integral (Range.linear (0 :: Natural) 20)
+  let pvMax = pvMin + pvDiff
+  pvAlt  <- Gen.integral (Range.linear (0 :: Natural) 100)
+  return $ ProtVer pvMin pvMax pvAlt
+
+-- | Generates a Double in the inclusive range [0, 1]
```

Comment:
Why not using [`double`](http://hackage.haskell.org/package/hedgehog-1.0/docs/Hedgehog-Gen.html#v:double)?

Reply:
Because its upper bound is exclusive, while I need it inclusive here.